e2e/msa/addKeyLowCost.ethereum.test.ts

```diff
@@ -0,0 +1,123 @@
+    return { delegatorKeys, addKeyPayload, delegatorSig, newSig };
+  }
+
+  // create AddKeyData using the provided keys.
+  // use the keys to sign the AddKeyData in an AddKey payload.
+  // return the new keys, the payload, and both signatures.
+  async function generateAddKeyCallParamsUsingKeys(
+    delegatorKeys: KeyringPair,
+    delegatorMsaId: u64,
+    ethereumKeyringPair: KeyringPair
+  ) {
+    const ethereumKeyPair = getEthereumKeyPairFromUnifiedAddress(getUnifiedAddress(ethereumKeyringPair));
+    const addKeyPayload = await generateAddKeyPayload({});
+    addKeyPayload.msaId = delegatorMsaId;
+    addKeyPayload.newPublicKey = getUnifiedPublicKey(ethereumKeyringPair);
+    const addKeyData = ExtrinsicHelper.api.registry.createType('PalletMsaAddKeyData', addKeyPayload);
+
+    const delegatorSig = signPayloadSr25519(delegatorKeys, addKeyData);
+    const newSig = await signEip712AddKeyData(ethereumKeyPair, addKeyPayload);
+    return { addKeyPayload, delegatorSig, newSig };
+  }
+
+  it('addPublicKeyToMsa costs less for capacity call with eligibility conditions', async function () {
+    // SET UP
+    const { delegatorKeys, addKeyPayload, delegatorSig, newSig } = await generateAddKeyCallParams();
+
+    // the extrinsic will be called by a provider with stake.
+    const addPublicKeyOp = new Extrinsic(
+      () =>
+        ExtrinsicHelper.api.tx.msa.addPublicKeyToMsa(
+          getUnifiedPublicKey(delegatorKeys),
+          delegatorSig,
+          newSig,
+          addKeyPayload
+        ),
+      providerKeys,
+      ExtrinsicHelper.api.events.msa.PublicKeyAdded
+    );
+
+    // ACT  pay with capacity using the provider.
+    const { eventMap } = await addPublicKeyOp.payWithCapacity();
+
+    // ASSERT it's a very small fee, but not free.
+    assertEvent(eventMap, 'msa.PublicKeyAdded');
+    const capacityFee = ExtrinsicHelper.getCapacityFee(eventMap);
+    assert(capacityFee > 0);
+    assert(capacityFee < 4n * CENTS);
+
+    // add another key; this should cost a lot more
+    const thirdKeyEth = await createKeys('Eth2', 'ethereum');
+    const delegatorMsaId: MessageSourceId = addKeyPayload.msaId || new u64(ExtrinsicHelper.api.registry, 0);
+    const newParams = await generateAddKeyCallParamsUsingKeys(delegatorKeys, delegatorMsaId, thirdKeyEth);
+
+    // again to be submitted by provider.
+    const addThirdKeyOp = new Extrinsic(
+      () =>
+        ExtrinsicHelper.api.tx.msa.addPublicKeyToMsa(
+          getUnifiedPublicKey(delegatorKeys),
+          newParams.delegatorSig,
+          newParams.newSig,
+          newParams.addKeyPayload
+        ),
+      providerKeys,
+      ExtrinsicHelper.api.events.msa.PublicKeyAdded
+    );
+
+    // ACT pay with capacity to add a third key
+    const { eventMap: eventMap2 } = await addThirdKeyOp.payWithCapacity();
+    assertEvent(eventMap2, 'msa.PublicKeyAdded');
+    const thirdKeyCapacityFee = ExtrinsicHelper.getCapacityFee(eventMap2);
+    // 4260363n vs
+    // 3258176n
```

Comment:
This doesn't seem like much of a discount... but look at the Weight returned [here](https://github.com/frequency-chain/frequency/pull/2403/files#diff-4a481e841703b9d88a0508a5d2e834def9a2fa7bcea617cdfc4243796067e82cR466)